rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -500,15 +553,15 @@ void debugger_frame::UpdateUnitList()
 	//const int old_size = m_choice_units->count();
 	QVariant old_cpu = m_choice_units->currentData();
 
+	bool reselected = false;
+
 	const auto on_select = [&](u32 id, cpu_thread& cpu)
 	{
 		if (emu_state == system_state::stopped) return;
 
-		QVariant var_cpu = QVariant::fromValue<std::weak_ptr<cpu_thread>>(
-			id >> 24 == 1 ? static_cast<std::weak_ptr<cpu_thread>>(idm::get_unlocked<named_thread<ppu_thread>>(id)) : idm::get_unlocked<named_thread<spu_thread>>(id));
-
-		m_choice_units->addItem(qstr(cpu.get_name()), var_cpu);
-		if (old_cpu == var_cpu) m_choice_units->setCurrentIndex(m_choice_units->count() - 1);
+		QVariant var_cpu = QVariant::fromValue<std::pair<cpu_thread*, u32>>(std::make_pair(&cpu, id));
+		m_choice_units->addItem(qstr(id >> 24 == 0x55 ? "RSX[0x55555555]" : cpu.get_name()), var_cpu);
+		if (!reselected && old_cpu == var_cpu) m_choice_units->setCurrentIndex(m_choice_units->count() - 1), reselected = true;
```

Comment:
scope this please. no need to squeeze the code like this

Reply:
Fixed.